bootstrap/handlers/secret/client/vaultclient_test.go

```diff
@@ -0,0 +1,274 @@
+			expectedNilCallback: false,
+			expectedNewToken:    "replacement-token",
+			expectedRetry:       true,
+			expectError:         true,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with expired token, non-existing TokenFile path",
+			authToken:           "expiredToken",
+			tokenFile:           "testdata/non-existing.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         true,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with expired test token, but same expired replacement token",
+			authToken:           "test-token",
+			tokenFile:           "testdata/testToken.json",
+			expectedNilCallback: false,
+			expectedNewToken:    "test-token",
+			expectedRetry:       false,
+			expectError:         true,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with unauthenticated token",
+			authToken:           "test-token",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         true,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with unrenewable token",
+			authToken:           "unrenewableToken",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       true,
+			expectError:         false,
+			expectedErrorType:   nil,
+		},
+		{
+			name:                "New secret client with no TokenFile",
+			authToken:           "testToken1",
+			tokenFile:           "",
+			expectedNilCallback: true,
+			expectedNewToken:    "",
+			expectedRetry:       false,
+			expectError:         false,
+			expectedErrorType:   nil,
+		},
+	}
+
+	for _, test := range tests {
+		testSecretStoreInfo.TokenFile = test.tokenFile
+		// pinned local variables to avoid scopelint warnings
+		testToken := test.authToken
+		cfgHTTP := vault.SecretConfig{
+			Host:           host,
+			Port:           portNum,
+			Protocol:       "http",
+			Authentication: vault.AuthenticationInfo{AuthToken: testToken},
+		}
+		expectError := test.expectError
+		expectedErrorType := test.expectedErrorType
+		expectedCallbackNil := test.expectedNilCallback
+		expectedNewToken := test.expectedNewToken
+		expectedRetry := test.expectedRetry
+
+		t.Run(test.name, func(t *testing.T) {
+			sclient := NewVault(bkgCtx, cfgHTTP, lc)
+			_, err := sclient.Get(testSecretStoreInfo)
+
+			if expectedErrorType != nil && err == nil {
+				t.Errorf("Expected error %v but none was received", expectedErrorType)
+			}
+
+			if !expectError && err != nil {
+				t.Errorf("Unexpected error: %s", err.Error())
+			}
+
+			if expectError && expectedErrorType != nil && err != nil {
+				eet := reflect.TypeOf(expectedErrorType)
+				aet := reflect.TypeOf(err)
+				if !aet.AssignableTo(eet) {
+					t.Errorf("Expected error of type %v, but got an error of type %v", eet, aet)
+				}
```

Comment:
Apologies for not catching this the first time around.  Strongly discourage reflection in unit tests (or mostly anywhere for that matter).  Not sure what you're testing for here that requires reflection.  Generally speaking you shouldn't be reflecting on an implementations internals (for testing or for any other purpose) as it breaks encapsulation.

Reply:
ok, will remove it